### PR TITLE
Ask User Before Deleting Chaindata

### PR DESCRIPTION
**Pull request by nisdas**
This asks the user to confirm before deleting all the chain data on disk.

### CODE REVIEW

File: beacon-chain/node/node.go [hunk 1]
```diff
@@ -194,19 +196,48 @@ func (b *BeaconNode) Close() {
...
+
+		for {
+			fmt.Print(">> ")
+
+			line, _, err := reader.ReadLine()
+			if err != nil {
+				return err
+			}
+			lineInput := strings.ToUpper(string(line))
+			if lineInput != "Y" && lineInput != "N" {
```
Comment (shayzluf) [inline beacon-chain/node/node.go]: maybe it can be nice to trim before

File: beacon-chain/node/node.go [hunk 2]
```diff
@@ -194,19 +196,49 @@ func (b *BeaconNode) Close() {
...
 	baseDir := ctx.GlobalString(cmd.DataDirFlag.Name)
 	dbPath := path.Join(baseDir, beaconChainDBName)
 	d, err := db.NewDB(dbPath)
 	if err != nil {
 		return err
 	}
 	if b.ctx.GlobalBool(cmd.ClearDB.Name) {
-		if err := d.ClearDB(); err != nil {
-			return err
+		reader := bufio.NewReader(os.Stdin)
```
Comment (terencechain) [inline beacon-chain/node/node.go]: how about refactor this into its own function?  Something like `askToDelete` or better ?

File: beacon-chain/node/node.go [hunk 3]
```diff
@@ -194,19 +196,49 @@ func (b *BeaconNode) Close() {
...
+
+		log.Warn("This will delete all the chain data stored in your data directory. " +
+			"Do you want to proceed ?(Y/N)")
```
Comment (0xKiwi) [inline beacon-chain/node/node.go]: Change the spacing here to "Do you want to proceed? (Y/N)"

File: beacon-chain/node/node.go [hunk 4]
```diff
@@ -194,19 +196,49 @@ func (b *BeaconNode) Close() {
...
+			if lineInput != "Y" && lineInput != "N" {
+				log.Errorf("Invalid option of %s chosen, enter Y/N", line)
+				continue
+			}
+			if lineInput == "Y" {
+				log.Warn("Deleting all chain data from data directory")
+				clearDB = true
+				break
+			}
+			log.Info("Not deleting chain data, the db will be initialized" +
```
Comment (0xKiwi) [inline beacon-chain/node/node.go]: Can you also put periods at the ends of these?

**Pull request merged**